renku/core/utils/scm.py

```diff
@@ -51,25 +51,35 @@ def git_unicode_unescape(s: str, encoding: str = "utf-8") -> str:
     return s
 
 
-def shorten_message(message, max_length=100, cut=True):
-    """Shortens or wraps a commit message to be at most `max_len` characters per line."""
+def shorten_message(message: str, line_length: int = 100, body_length: int = 65000):
+    """Wraps and shortens a commit message.
 
-    if len(message) < max_length:
-        return message
+    :param message: message to adjust.
+    :param line_length: maximum line length before wrapping. 0 for infinite.
+    :param body_length: maximum body length before cut. 0 for infinite.
+    :return: message wrapped and trimmed.
+    """
+
+    if body_length and len(message) > body_length:
+        message = message[: body_length - 3] + "..."
 
-    if cut:
-        return message[: max_length - 3] + "..."
+    if not line_length or len(message) <= line_length:
```

Comment:
if `line_length` is allowed to be None, the type above should be `Optional[int]`

Reply:
I think `line_length != 0` (or `<1`) and keeping the `[int]` type may be better in this case.